bitwarden_license/test/Commercial.Core.Test/Services/ProviderServiceTests.cs

```diff
@@ -450,6 +450,53 @@ await sutProvider.GetDependency<IEventService>()
                 EventType.ProviderOrganization_Added);
     }
 
+    [Theory, BitAutoData]
+    public async Task AddOrganizationsToReseller_WithResellerProvider_Success(Provider provider, ICollection<Organization> organizations, SutProvider<ProviderService> sutProvider)
+    {
+        provider.Type = ProviderType.Reseller;
+        sutProvider.GetDependency<IProviderRepository>().GetByIdAsync(provider.Id).Returns(provider);
+
+        var providerOrganizationRepository = sutProvider.GetDependency<IProviderOrganizationRepository>();
+        foreach (var organization in organizations)
+        {
+            organization.PlanType = PlanType.EnterpriseAnnually;
+            providerOrganizationRepository.GetByOrganizationId(organization.Id).ReturnsNull();
+            sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organization.Id).Returns(organization);
+        }
+
+        var organizationIds = organizations.Select(o => o.Id).ToArray();
+
+        await sutProvider.Sut.AddOrganizationsToReseller(provider.Id, organizationIds);
+
+        await providerOrganizationRepository.ReceivedWithAnyArgs().CreateWithManyOrganizations(Arg.Is<ProviderOrganization>(p => p.ProviderId == provider.Id), organizationIds);
+        await sutProvider.GetDependency<IEventService>()
+            .Received().LogProviderOrganizationEventsAsync(provider.Id, Arg.Any<IEnumerable<ProviderOrganization>>(),
```

Comment:
Please use `Arg.Is` instead of `Arg.Any` so that we know it's been called correctly.